sqlglot/dialects/dialect.py

```diff
@@ -110,6 +124,9 @@ class Dialect(metaclass=_Dialect):
     parser_class = None
     generator_class = None
 
+    def __eq__(self, other: DialectType) -> bool:  # type: ignore
+        return type(self) == other
```

Comment:
why do you need this one? shouldn't the metaclass wokr?

Reply:
Nope, without it:

```python
AssertionError: <sqlglot.dialects.snowflake.Snowflake object at 0x125740700> != 'snowflake'
```